### PR TITLE
OCD-2775- Improve Reports responsiveness - save filter

### PR DESCRIPTION
**Pull request by tmy1313**
fix: update listings report to switch between filters with productIds
fix: refresh filter list when user stops impersonating

### CODE REVIEW

File: src/app/pages/reports/listings.component.js [hunk 1]
```diff
@@ -29,6 +30,11 @@ export const ReportsListingsComponent = {
             if (changes.productId && changes.productId.currentValue) {
                 this.productId = angular.copy(changes.productId.currentValue);
             }
+            if (changes.filterToApply && changes.filterToApply.currentValue && changes.filterToApply.currentValue !== changes.filterToApply.previousValue) {
```
Comment (andlar) [inline src/app/pages/reports/listings.component.js]: if "filterToApply" is in the changes object, then that means it changed. Also, if filterToApply is an object, you shouldn't use `===` or `!==` to test for equality

File: src/app/pages/reports/reports.state.js [hunk 2]
```diff
@@ -22,9 +22,11 @@ function reportsStateConfig ($stateProvider) {
             component: 'chplReportsListings',
             params: {
                 productId: {squash: true, value: null},
+                filterToApply: {squash: true, value: null},
```
Comment (andlar) [inline src/app/pages/reports/reports.state.js]: I don't think this should be here. The things in params are part of the URL

File: src/app/pages/reports/reports.state.js [hunk 3]
```diff
@@ -22,9 +22,11 @@ function reportsStateConfig ($stateProvider) {
...
             },
             resolve: {
                 productId: $transition$ => $transition$.params().productId,
+                filterToApply: $transition$ => $transition$.params().filterToApply,
```
Comment (andlar) [inline src/app/pages/reports/reports.state.js]: Can you try this without this line? I'm not sure if it's needed

File: src/app/pages/reports/reports.state.js (hunk 2 above)
Comment (tmy1313) [reply src/app/pages/reports/reports.state.js]: I tried removing each of the references to `filterToApply`, but if they are both not present, the `filterToApply` value is not in the `changes` object.  It seems they re both needed.

File: src/app/pages/reports/reports.state.js (hunk 3 above)
Comment (tmy1313) [reply src/app/pages/reports/reports.state.js]: I tried removing each of the references to `filterToApply`, but if they are both not present, the `filterToApply` value is not in the `changes` object.  It seems they re both needed.

**Pull request merged**